test/e2e/density.go

```diff
@@ -0,0 +1,191 @@
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  name,
+							Image: image,
+							Ports: []api.ContainerPort{{ContainerPort: 80}},
+						},
+					},
+				},
+			},
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	By(fmt.Sprintf("Making sure all %d replicas exist", replicas))
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	pods, err := c.Pods(ns).List(label)
+	Expect(err).NotTo(HaveOccurred())
+	current = len(pods.Items)
+	failCount := 5
+	for same < failCount && current < replicas {
+		Logf("Controller %s: Found %d pods out of %d", name, current, replicas)
+		if last < current {
+			same = 0
+		} else if last == current {
+			same++
+		} else if current < last {
+			Failf("Controller %s: Number of submitted pods dropped from %d to %d", last, current)
+		}
+
+		if same >= failCount {
+			Logf("No pods submitted for the last %d checks", failCount)
+		}
+
+		last = current
+		time.Sleep(5 * time.Second)
+		pods, err = c.Pods(ns).List(label)
+		Expect(err).NotTo(HaveOccurred())
+		current = len(pods.Items)
+	}
+	Expect(current).To(Equal(replicas))
+	Logf("Controller %s: Found %d pods out of %d", name, current, replicas)
+
+	By("Waiting for each pod to be running")
+	same = 0
+	last = 0
+	failCount = 6
+	unknown := 0
+	pending := 0
+	current = 0
+	for same < failCount && current < replicas {
+		current = 0
+		pending = 0
+		unknown = 0
+		time.Sleep(10 * time.Second)
+		for _, pod := range pods.Items {
+			p, err := c.Pods(ns).Get(pod.Name)
+			Expect(err).NotTo(HaveOccurred())
+			if p.Status.Phase == api.PodRunning {
+				current++
+			} else if p.Status.Phase == api.PodPending {
+				pending++
+			} else if p.Status.Phase == api.PodUnknown {
+				unknown++
+			}
+		}
+		Logf("Pod States: %d running, %d pending, %d unknown ", current, pending, unknown)
+		if last < current {
+			same = 0
+		} else if last == current {
+			same++
+		} else if current < last {
+			Failf("Number of running pods dropped from %d to %d", last, current)
+		}
+		if same >= failCount {
+			Logf("No pods started for the last %d checks", failCount)
+		}
+		last = current
+	}
+	Expect(current).To(Equal(replicas))
+}
+
+# This test suite can take a long time to run, so by default it is disabled
+# by being marked as Pending.  To enable this suite, remove the P from the
+# front of PDescribe (PDescribe->Describe) and then all tests will
+# be available
+var _ = PDescribe("Density", func() {
+	var c *client.Client
+	var minionCount int
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+		minions, err := c.Nodes().List()
+		expectNoError(err)
+		minionCount = len(minions.Items)
+		Expect(minionCount).NotTo(BeZero())
+	})
+
+	AfterEach(func() {
+		// Remove any remaining pods from this test
+		podClient := c.Pods(api.NamespaceDefault)
+		pods, _ := podClient.List(labels.Everything())
+		for _, pod := range pods.Items {
+			podClient.Delete(pod.Name)
```

Comment:
Please use the DeleteRC function-- deleting all pods like this will probably get rid of pods that should be running.